src/Client/Campaign.php

```diff
@@ -107,29 +108,51 @@ class Campaign extends Common
      * @param string $id
      * @return SFCampaignApiResponse Single Campaign response object as associative array
      * @throws NotFoundException if Campaign with given ID not found
+     * @throws TransferException if connection or request fails twice
+     * @throws \Exception if our loop logic or Guzzle has an unexpexcted case we missed
      */
     public function getById(string $id, bool $withCache): array
```

Comment:
retry loop looks good, although afaik at when MAT-405 is done the only reason we'll have to ever fetch a campaign from SF is to show the early preview, which is not that critical and might not need a retry system. No harm in having it though.

Reply:
Yeah not critical, but I think it's probably good to keep charity previews as stable as we can.